File: directives/sources/invalid_compound_names.1.f90
! @@name:       invalid_compound_names.1
! @@type:       F-free
! @@operation:  compile
! @@expect:     ct-error
! @@version:    omp_6.0
subroutine component_names(n, x)
  implicit none
  integer, intent(in)    :: n
  real,    intent(inout) :: x(n)
  integer                :: i
  interface
    subroutine f()
    end subroutine
    subroutine f(xi)
      real :: xi
    end subroutine
  end interface

  !! invalid: repeated parallel disallowed
  !$omp parallel target parallel
    call f()
  !$omp end parallel target parallel

  !! invalid: target cannot be nested inside teams
  !$omp teams target
    call f()
  !$omp end teams target

  !! invalid: sections may only be appended to parallel
  !$omp teams sections
    call f()
  !$omp end teams sections

  !! invalid: masked may only be appended to parallel
  !$omp teams masked
    call f()
  !$omp end teams masked

  !! invalid: multiple data-mapping constituents
  !$omp target_data parallel target map(x[:n])
    call f()
  !$omp end target_data parallel target map(x[:n])

  !! invalid: task and taskloop both generate tasks in
  !! same parallel region
  !$omp task taskloop
  do i = 1, n
    call g(x(i))
  end do

end subroutine

File: directives/sources/invalid_compound_names.1.c
/*
* @@name:       invalid_compound_names.1
* @@type:       C   
* @@operation:  compile
* @@expect:     ct-error
* @@version:    omp_6.0
*/
void f();
void g(float);

void invalid_compound_names(int n, float *x)
{

  // invalid: repeated parallel disallowed
  #pragma omp parallel target parallel
  f();

  // invalid: target cannot be nested inside teams
  #pragma omp teams target
  f();

  // invalid: sections may only be appended to parallel
  #pragma omp teams sections
  f();

  // invalid: masked may only be appended to parallel
  #pragma omp teams masked
  f();

  // invalid: multiple data-mapping constituents
  #pragma omp target_data parallel target map(x[:n])
  f();

  // invalid: task and taskloop both generate tasks in
  // same parallel region
  #pragma omp task taskloop
  for (int i = 0; i < n; i++) {
    g(x[i]);
  }
}
